Objects/dictobject.c

```diff
@@ -1629,6 +1629,15 @@ insert_into_splitdictkeys(PyDictKeysObject *keys, PyObject *name)
     return ix;
 }
 
+static Py_ssize_t
+insert_into_splitdictkeys(PyDictKeysObject *keys, PyObject *name)
+{
+    LOCK_KEYS(keys);
```

Comment:
Up to you, but the previous structure where `LOCK_KEYS()`/`UNLOCK_KEYS()` was in `_PyObject_StoreInstanceAttribute` seemed a bit clearer to me.